session.go

```diff
@@ -92,6 +90,73 @@ func (s *Session) init(opts *SessionOptions) {
 	}
 	// create handle map after options have been applied
 	s.handles = bitmap.New(s.handleMax)
+	return s
+}
+
+func (s *Session) begin(ctx context.Context) error {
+	// send Begin to server
+	begin := &frames.PerformBegin{
+		NextOutgoingID: 0,
+		IncomingWindow: s.incomingWindow,
+		OutgoingWindow: s.outgoingWindow,
+		HandleMax:      s.handleMax,
+	}
+	debug.Log(1, "TX (NewSession): %s", begin)
+
+	// we use send to have positive confirmation on transmission
+	send := make(chan encoding.DeliveryState)
+	_ = s.txFrame(begin, send)
```

Comment:
We probably shouldn't ignore this error - if the connection is in a bad state bailing out earlier is probably best.

Reply:
This is another bit of cruft I intend to clean up, but not in the way you might think.

`Session.txFrame()` calls `conn.SendFrame()` and the error returned is if `conn` has terminated.  i.e. it checks `<-conn.Done` and when closed it returns `conn.Err()`.  This is redundant, as when sending a frame and waiting for an ack we always wait on `<-conn.Done`.

I believe that we can update `Session.txFrame()` to not return an error.  However, I need to audit all of the call-sites to be sure.  If you search through the codebase, you'll find `_ = s.txFrame(...)` all over due to this pattern.

Reply:
Ah, I see it. Makes sense.